### PR TITLE
feat(combobox): add invalid state for combobox

### PR DESCRIPTION
**Pull request by Donisius**

### CODE REVIEW

File: src/combobox/combobox.component.ts [hunk 1]
```diff
@@ -86,6 +91,7 @@ import { filter } from "rxjs/operators";
 						ariaLabel="Close menu">
 					</ibm-icon-chevron-down16>
 			</div>
+			<div *ngIf="invalid" class="bx--form-requirement">A valid value is required</div>
```
Comment (Pablodotnet) [inline src/combobox/combobox.component.ts]: invalidText input can be integrated to as in dropdowns or inputs to customize this invalid message

Comment (cal-smith) [reply src/combobox/combobox.component.ts]: 💯 agreed, there should be an `invalidText` input like in `dropdown.component`

File: src/combobox/combobox.component.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ import { filter } from "rxjs/operators";
 		</div>
 		<div
 			class="bx--combo-box bx--list-box"
+			[attr.data-invalid]="invalid ? false : null"
```
Comment (zvonimirfras) [inline src/combobox/combobox.component.ts]: Shouldn't it be
```suggestion
			[attr.data-invalid]="(invalid ? true : null)"
```

File: src/combobox/combobox.component.ts [hunk 3]
```diff
@@ -151,6 +158,14 @@ export class ComboBox implements OnChanges, AfterViewInit, AfterContentInit {
...
 	 */
 	@Input() helperText: string | TemplateRef<any>;
+	/**
+	 * Set to `true` for invalid state.
+	 */
+	@Input() invalid = false;
+	/**
+	 * Value displayed if dropdown is in invalid state.
+	 */
+	@Input() invalidText = "";
```
Comment (zvonimirfras) [inline src/combobox/combobox.component.ts]: We usually eventually need string or template options.
Any reason why that won't be the case here?

File: src/combobox/combobox.component.ts (hunk 2 above)
Comment (Donisius) [reply src/combobox/combobox.component.ts]: I agree, it should be like that.

File: src/combobox/combobox.component.ts [hunk 4]
```diff
@@ -92,6 +97,8 @@ import { filter } from "rxjs/operators";
 				<ng-content></ng-content>
 			</div>
 		</div>
+
+		<div *ngIf="invalid" class="bx--form-requirement">{{ invalidText }}</div>
```
Comment (zvonimirfras) [inline src/combobox/combobox.component.ts]: Template also needs a template outlet to be displayed :)

Combo box does it for example.

Comment (Donisius) [reply src/combobox/combobox.component.ts]: Oh right 🙃 , I updated it to include the outlet and an example in the storybook

**Pull request merged**